### PR TITLE
Use the new layout system in ContentPresenter

### LINKED ISSUE #7805: Root Grid in a ContentView with a ControlTemplate with a ContentPResenter does not size
### Description

If a Grid is the root element in the Content of a ContentView (or TemplatedView) then it does not resize (other root views work):

```xaml
<ContentView>
    <ContentView.ControlTemplate>
        <ControlTemplate>
            <ContentPresenter Content="{Binding Content, Source={RelativeSource TemplatedParent}}" />
        </ControlTemplate>
    </ContentView.ControlTemplate>
    <Grid>
        <Button />
    </Grid>
</ContentView>
```

### Steps to Reproduce

1. Paste the snippet in a page
2. Run the app and observe the button fills the screen
4. Resize the window
5. Observe the button does not resize
6. Change the root Grid to a VerticalStackLayout / Border / StackLayout
7. Run the app again and observe the button resize

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

All

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

We need to call the new layout system with the ContentPresenter because it is still using the compatibility layouts as a base. This same logic is applied to ContentView, ContentPage and TemplatedView (base of ContentView and RadioButton).

### Issues Fixed

Fixes #7805

### CODE REVIEW

File: src/Controls/src/Core/PublicAPI/net6.0-maccatalyst/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -1,4 +1,7 @@
 ﻿#nullable enable
+override Microsoft.Maui.Controls.ContentPresenter.ArrangeOverride(Microsoft.Maui.Graphics.Rect bounds) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentPresenter.MeasureOverride(double widthConstraint, double heightConstraint) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentView.OnApplyTemplate() -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/net6.0-maccatalyst/PublicAPI.Unshipped.txt:4]:
```suggestion
```

File: src/Controls/src/Core/PublicAPI/net6.0-tizen/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ﻿#nullable enable
+override Microsoft.Maui.Controls.ContentPresenter.ArrangeOverride(Microsoft.Maui.Graphics.Rect bounds) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentPresenter.MeasureOverride(double widthConstraint, double heightConstraint) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentView.OnApplyTemplate() -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/net6.0-tizen/PublicAPI.Unshipped.txt:4]:
```suggestion
```

File: src/Controls/src/Core/PublicAPI/net6.0-windows10.0.19041/PublicAPI.Unshipped.txt [hunk 3]
```diff
@@ -1,4 +1,7 @@
 ﻿#nullable enable
+override Microsoft.Maui.Controls.ContentPresenter.ArrangeOverride(Microsoft.Maui.Graphics.Rect bounds) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentPresenter.MeasureOverride(double widthConstraint, double heightConstraint) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentView.OnApplyTemplate() -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/net6.0-windows10.0.19041/PublicAPI.Unshipped.txt:4]:
```suggestion
```

File: src/Controls/src/Core/PublicAPI/net6.0/PublicAPI.Unshipped.txt [hunk 4]
```diff
@@ -1,4 +1,7 @@
 ﻿#nullable enable
+override Microsoft.Maui.Controls.ContentPresenter.ArrangeOverride(Microsoft.Maui.Graphics.Rect bounds) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentPresenter.MeasureOverride(double widthConstraint, double heightConstraint) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentView.OnApplyTemplate() -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/net6.0/PublicAPI.Unshipped.txt:4]:
```suggestion
```

File: src/Controls/src/Core/PublicAPI/netstandard2.0/PublicAPI.Unshipped.txt [hunk 5]
```diff
@@ -1,4 +1,7 @@
 ﻿#nullable enable
+override Microsoft.Maui.Controls.ContentPresenter.ArrangeOverride(Microsoft.Maui.Graphics.Rect bounds) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentPresenter.MeasureOverride(double widthConstraint, double heightConstraint) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentView.OnApplyTemplate() -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/netstandard2.0/PublicAPI.Unshipped.txt:4]:
```suggestion
```

File: src/Controls/src/Core/PublicAPI/netstandard2.1/PublicAPI.Unshipped.txt [hunk 6]
```diff
@@ -1,4 +1,7 @@
 ﻿#nullable enable
+override Microsoft.Maui.Controls.ContentPresenter.ArrangeOverride(Microsoft.Maui.Graphics.Rect bounds) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentPresenter.MeasureOverride(double widthConstraint, double heightConstraint) -> Microsoft.Maui.Graphics.Size
+override Microsoft.Maui.Controls.ContentView.OnApplyTemplate() -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/netstandard2.1/PublicAPI.Unshipped.txt:4]:
```suggestion
```

**Comment by rmarinho** [comment]
Shouldn0t this be target to main ?

**Review by hartez — APPROVED** [review]

**Pull request merged**